src/data/list/basic.lean

```diff
@@ -3480,6 +3480,60 @@ begin
     { simp [hmem] } }
 end
 
+@[simp] lemma enum_nil : enum ([] : list α) = [] := rfl
+@[simp] lemma enum_from_nil (n : ℕ) : enum_from n ([] : list α) = [] := rfl
+
+lemma enum_from_cons (x : α) (xs : list α) (n : ℕ) :
+  enum_from n (x :: xs) = (n, x) :: enum_from (n + 1) xs := rfl
+lemma enum_cons (x : α) (xs : list α) :
+  enum (x :: xs) = (0, x) :: enum_from 1 xs := enum_from_cons _ _ _
+@[simp] lemma enum_from_singleton (x : α) (n : ℕ) :
+  enum_from n [x] = [(n, x)] := enum_from_cons _ _ _
+@[simp] lemma enum_singleton (x : α) :
+  enum [x] = [(0, x)] := enum_cons _ _
+
+lemma enum_from_append (xs ys : list α) (n : ℕ) :
+  enum_from n (xs ++ ys) = enum_from n xs ++ enum_from (n + xs.length) ys :=
+begin
+  induction xs with x xs IH generalizing ys n,
+  { simp },
+  { rw [cons_append, enum_from_cons, IH, ←cons_append, ←enum_from_cons,
+        length, add_right_comm, add_assoc] }
+end
+
+lemma enum_append (xs ys : list α) :
+  enum (xs ++ ys) = enum xs ++ enum_from xs.length ys :=
+by simp [enum, enum_from_append]
+
+lemma map_fst_add_enum_eq_enum_from (l : list α) (n : ℕ) :
+  map (prod.map (+ n) id) (enum l) = enum_from n l :=
+begin
+  rw [enum],
+  induction l with hd tl IH generalizing n,
+  { simp [enum_from] },
+  { simp only [enum_from, map, zero_add, prod.map_mk, id.def,
+               eq_self_iff_true, true_and],
+    rw [←IH (n + 1), ←IH 1, map_map],
+    congr,
+    ext;
+    simp [add_comm, add_left_comm] }
+end
+
+lemma nth_le_enum_from (l : list α) (n i : ℕ)
```

Comment:
Should these two lemmas be `simp`? I don't know.

Reply:
They aren't for things like `nth_le_append`, `nth_le_map`, `nth_le_take`.